### PR TITLE
init backend files

### PR DESCRIPTION
**Pull request by amirg76**
please check messages and categories features

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { connectDB } from './services/db.js';
+import app from './app.js';
+import AppError from './errors/AppError.js';
+import logger from './services/logger.js';
+
+config({ path: 'src/config/' });
+connectDB()
+  .then(() => console.log('DB connected!'))
+  .catch(() => {
+    throw new AppError('Database connection error', 500, true);
```
Comment (AmeerLalaDeveloper) [inline src/index.js:12]: try not to hardcode the status codes and use the the constants from the errorManagment object

File: src/services/db.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+export async function connectDB() {
+  try {
+    await mongoose.connect(getMongoUri());
+
+    mongoose.connection.on('error', (err) => {
+      throw new AppError('Database connection error', 500, true);
+    });
+
+    mongoose.connection.on('disconnected', () => {
+      throw new AppError('Database disconnected', 500, true);
```
Comment (AmeerLalaDeveloper) [inline src/services/db.js:14]: try not to hardcode the status codes and use the the constants from the errorManagment object

File: src/services/db.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    });
+  } catch (_) {
+    throw new AppError('Database connection error', 500, true);
```
Comment (AmeerLalaDeveloper) [inline src/services/db.js:17]: try not to hardcode the status codes and use the the constants from the errorManagment object

File: src/errors/catchProgrammingErrors.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  } catch (err) {
+    if (err instanceof AppError) {
+      next(err);
+    } else if (err instanceof Error) {
+      // Check if err is an instance of Error
+      // If it's a programmer error, wrap it in an AppError and set it as a programming error
+      next(new AppError(err.message, 500, false));
+    } else {
+      // Handle other unknown errors here
+      next(new AppError('Unknown error', 500, false));
```
Comment (AmeerLalaDeveloper) [inline src/errors/catchProgrammingErrors.js:15]: try not to hardcode the status codes and use the the constants from the errorManagment object

File: src/errors/utils/dbErrorHandling.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import AppError from '../AppError.js'; // Update the path as needed
+
+export function applyErrorHandlingMiddleware(schema) {
+  schema.post('save', (error, doc, next) => {
+    if (error) {
+      if (error.name === 'ValidationError') {
+        next(new AppError('Validation error', 400, true));
```
Comment (AmeerLalaDeveloper) [inline src/errors/utils/dbErrorHandling.js:7]: try not to hardcode the status codes and use the the constants from the errorManagment object

File: src/errors/utils/dbErrorHandling.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+      } else if (error.name === 'MongoError' && error.code === 11000) {
+        next(new AppError('Duplicate key error', 409, true));
```
Comment (AmeerLalaDeveloper) [inline src/errors/utils/dbErrorHandling.js:9]: try not to hardcode the status codes and use the the constants from the errorManagment object

File: src/errors/catchProgrammingErrors.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import AppError from './AppError.js'; // Adjust the import path as needed
+
+const catchProgrammingErrors = (handler) => async (req, res, next) => {
```
[the remark this answers is not in the document]
Comment (thesamirkhoury) [reply src/errors/catchProgrammingErrors.js:4]: This is what we got from the boilerplate code, is there any recommendations to fix/change this?

**Pull request merged**